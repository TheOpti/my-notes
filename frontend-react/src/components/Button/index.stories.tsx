import * as React from 'react';
import { storiesOf } from '@storybook/react';

import Button from './';

const btnWrapperStyle = {
  margin: '20px',
};

storiesOf('Button', module)
  .add('all buttons', (): any => (
    <>
      <div style={btnWrapperStyle}>
        <Button label="Raised button" color="raised" />
      </div>
      <div style={btnWrapperStyle}>
        <Button label="Outlined button" color="outlined" />
      </div>
    </>
  ))
  .add('with text', () => (
    <Button label="This is sample button" />
  ));