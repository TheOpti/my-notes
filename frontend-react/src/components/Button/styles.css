.root {
  border: none;
  border-radius: 4px;
  padding: 0 16px;
  min-width: 64px;
  height: 36px;
  color: white;
  background-color: rgb(var(--pure-material-primary-rgb, 33, 150, 243));
  outline: none;
  cursor: pointer;
  transition: box-shadow 0.3s, opacity 0.3s;
  box-shadow: 0 3px 1px -2px rgba(0, 0, 0, 0.2), 0 2px 2px 0 rgba(0, 0, 0, 0.14), 0 1px 5px 0 rgba(0, 0, 0, 0.12);
}

.root:hover {
  box-shadow: 
    0 2px 4px -1px rgba(0, 0, 0, 0.2), 
    0 4px 5px 0 rgba(0, 0, 0, 0.14), 
    0 1px 10px 0 rgba(0, 0, 0, 0.12);
  opacity: 0.9;
}
