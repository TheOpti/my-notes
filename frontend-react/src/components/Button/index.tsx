import React from 'react';
import styles from './styles.css';

const Button = ({ label, onClickHandler }) => {
  return (
    <button 
      onClick={onClickHandler}
      className={styles.root}>
      { label }
    </button>
  )
}

export default Button;
