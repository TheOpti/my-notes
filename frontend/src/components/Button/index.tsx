import React from 'react';
import cx from 'classnames';
import styles from './styles.css';

type ButtonPropsType = {
  label: string;
  onClickHandler?: (event: any) => void;
  color?: 'raised' | 'outlined';
  classname?: string,
  disabled?: boolean,
}
const Button: React.SFC<ButtonPropsType> = (props: ButtonPropsType) => {
  const { label, onClickHandler, color, classname, disabled } = props;

  const btnClasses = cx(styles.root, styles[color], classname, {
    [styles.disabled]: disabled,
  });

  return (
    <button 
      onClick={onClickHandler}
      className={btnClasses}
    >
      { label }
    </button>
  )
}

Button.defaultProps = {
  label: '',
  color: 'raised',
  classname: '',
};

export default Button;
