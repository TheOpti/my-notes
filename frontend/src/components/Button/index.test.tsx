import React from "react";
import { shallow } from 'enzyme';
import Button from './';

describe('Button', () => {
  it('should render correctly without any issues', () => {
    const wrapper = shallow(<Button 
      label="Label" 
      onClickHandler={() => {}} 
      />
    );

    expect(wrapper).toMatchSnapshot();
  });
});