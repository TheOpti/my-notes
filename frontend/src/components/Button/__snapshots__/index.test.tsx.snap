// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render correctly without any issues 1`] = `
<Button
  classname=""
  color="raised"
  label="Label"
  onClickHandler={[Function]}
>
  <button
    className="root raised"
    onClick={[Function]}
    type="button"
  >
    <span
      style={Object {}}
    >
      Label
    </span>
  </button>
</Button>
`;
